### PR TITLE
- parser rewrite close to python source

### PR DESCRIPTION
**Pull request by PaddeK**
Just a rewrite of the apdu parser.. bit closer to the python source.
Should resolve your Todo "Use hipster(correct) JS declarations IE let instead of var".
No need for "Use async/await/promises not callbacks" because the code is not asynchronous.

I realized the response SW1/SW2 can contain non hex codes like "--" and "XX" this isn't handled
in any way. Even the python source ignores this.

Thanks for creating the Omni ring :)

--- Edit ---
I reorganized the source APDUs and now wildcard parts of reponse APDUs are handled.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,164 @@
+# Created by .ignore support plugin (hsz.mobi)
```
Comment (JohnMcLear) [inline .gitignore]: Afaik all this cruft should not be committed.

File: parseAPDU.js [hunk 2]
```diff
@@ -1,2256 +1,2737 @@
-function parseAPDU(apdu, callback){
+(() => {
```
Comment (JohnMcLear) [inline parseAPDU.js]: Please maintain the same line indentation, this is so I can review line by line and not have to review all content :)

**Comment by JohnMcLear** [comment]
Awesome start, thanks!  Sorry about the slow response. 

I put in a couple of review pointers I think are worth addressing.  .gitignore cruft is just pointless and should not be committed unless it's absolutely critical to your workflow.

I also feel that unnecessarily changing the indentation away from the original style makes code review too complex.  An option here if you absolutely need to change indentation is to make 1 pull request for logic then a separate PR for styling / indentation :)

Really happy to see this contribution land so whenever you can make the proposed changes I will get the PR merged :)

Thanks again!

**Comment by PaddeK** [comment]
Sorry i forgot about this.. vacation is over so no real spare time.
What do you think about moving the command/response APDU objects to seperate files in a first step.
The actual code is sub 100 lines and the 2 static objects bloat it about 25 times.

**Comment by JohnMcLear** [comment]
Yea that sounds super sensible +1 :D



-----Original message-----

**Comment by PaddeK** [comment]
First step is done.. tomorrow next steps

**Comment by JohnMcLear** [comment]
Nice one thanks!  Will b e offline for a bit so will make you admin to merge yourself

**Comment by JohnMcLear** [comment]
merged anyway so we can get things moving, web nfc now more of a thing so more important we get this out

**Pull request merged**